File: judge/libraries/PortalLib1.java
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////

public class PortalLib1 {
	public static final int NSPEEDS = 1000;
	
	public static int nLeftTeam;
	public static int nTopTeam;
	public static int nLeftJudge;
	public static int nTopJudge;
	
	public static void answer(int T, int L){
		nLeftTeam = L;
		nTopTeam = T;
	}
	
	public static void main(String args[]){
		int ntc, nsp;
		Cube c = new Cube(0,0);
		double speeds[];
		java.util.Scanner sc = new java.util.Scanner(System.in);
		
		ntc = sc.nextInt();
		for (int i=0; i<ntc; i++){
			speeds = new double[NSPEEDS];
			c.x = sc.nextInt(); c.y = sc.nextInt(); nsp = sc.nextInt();
			for (int j=0; j<nsp; j++){
				speeds[j] = sc.nextDouble();
			}
			nLeftJudge = sc.nextInt(); nTopJudge = sc.nextInt();
			Portal1.speed_assessment(c,nsp,speeds);
			if (nLeftTeam != nLeftJudge || nTopTeam != nTopJudge){
				System.out.printf("INCORRECT - Test case %d received (Left,top) (%d,%d) / expected (%d,%d)\n",
					i+1, nLeftTeam, nTopTeam, nLeftJudge, nTopJudge);
				System.exit(0);
			}
		}
		System.out.printf("CORRECT\n");
	}
}

File: judge/libraries/portalLib1.c
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////
#include<stdlib.h>
#include<string.h>
#include<stdio.h>

#include"portalLib1.h"

#define	NSPEEDS		1000

static int nGateTeam, nTopTeam, nGateJudge, nTopJudge;

void answer(int T, int L){
	nGateTeam = L;
	nTopTeam = T;
}

int main(){
	int ntc, nsp, i, j;
	cube c;
	double speeds[NSPEEDS];
	scanf("%d",&ntc);
	for (i=0; i<ntc; i++){
		memset(speeds,0,sizeof(speeds));
		scanf("%d %d %d",&c.x, &c.y, &nsp);
		for (j=0; j<nsp;j++){
			scanf("%lf",&speeds[j]);
		}
		scanf("%d %d", &nGateJudge, &nTopJudge);
		speed_assessment(c,nsp,speeds);
		if (nGateTeam != nGateJudge || nTopTeam != nTopJudge){
			printf("INCORRECT - Test case %d received (gate,top) (%d,%d) / expected (%d,%d)\n",
				i+1, nGateTeam, nTopTeam, nGateJudge, nTopJudge);
			exit(0);
		}
	}
	printf("OK - 6673454bb0eb6491f54407deed2b7a23\n");
	return 0;
}